<?xml version="1.0"?>
<launch>

  <include file="$(find self_drive_tasks_gazebo)/launch/q4_gazebo.launch" />

  <include file="$(find sequoia_twist_controller)/launch/sequoia_twist_controller.launch" />

  <node ns="camera_sim" pkg="sequoia_lane_detection" type="lane_detection_node" name="lane_detection_node" output="screen" />

  <node pkg="lidar_stop_trigger" type="lidar_stop_trigger" name="lidar_stop_trigger" >
    <remap from="scan" to="/laser/scan_filtered" />
    <remap from="stop_trigger" to="vehicle/stop_trigger" />
  </node>

  <node pkg="task_nodes" type="multiplexor" name="multiplexor" output="screen" >
    <remap from="cmd_vel" to="/vehicle/cmd_vel" />
    <remap from="gear_cmd" to="/vehicle/gear_cmd" />
  </node>

  <node ns="laser" pkg="laser_filters" type="scan_to_scan_filter_chain" name="scan_filter" output="screen" respawn="true" >
    <rosparam file="$(find sequoia_nav_laser)/yaml/laser_filter_params.yaml" />
  </node>

  <node pkg="move_vehicle" type="move_vehicle" name="move_vehicle" output="screen">
  </node>

  <node pkg="move_base" type="move_base" name="move_base" output="screen" >
    <rosparam file="$(find sequoia_nav_laser)/yaml/move_base_params.yaml" />
    <rosparam file="$(find sequoia_nav_laser)/yaml/global_planner_params.yaml" ns="NavfnROS" />
    <rosparam file="$(find sequoia_nav_laser)/yaml/global_costmap_params_solid_line.yaml" ns="global_costmap" />
    <rosparam file="$(find sequoia_nav_laser)/yaml/local_costmap_params_solid_line.yaml" ns="local_costmap" />
    <rosparam file="$(find sequoia_nav_laser)/yaml/local_planner_params.yaml" ns="TebLocalPlannerROS" />
    <rosparam file="$(find sequoia_nav_laser)/yaml/costmap_converter_params.yaml" command="load" />
    <remap from="cmd_vel" to="cmd_vel_solid" />
  </node>

  <node pkg="move_base" type="move_base" name="move_base_dashed" output="screen" >
    <rosparam file="$(find sequoia_nav_laser)/yaml/move_base_params.yaml" />
    <rosparam file="$(find sequoia_nav_laser)/yaml/global_planner_params.yaml" ns="NavfnROS" />
    <rosparam file="$(find sequoia_nav_laser)/yaml/global_costmap_params_dashed_line.yaml" ns="global_costmap" />
    <rosparam file="$(find sequoia_nav_laser)/yaml/local_costmap_params_dashed_line.yaml" ns="local_costmap" />
    <rosparam file="$(find sequoia_nav_laser)/yaml/local_planner_params.yaml" ns="TebLocalPlannerROS" />
    <rosparam file="$(find sequoia_nav_laser)/yaml/costmap_converter_params.yaml" command="load" />
    <remap from="cmd_vel" to="cmd_vel_dashed" />
  </node>

  <node pkg="ekf_example" type="ekf_example" name="ekf_example"/>

  <rosparam file="$(find sequoia_nav_laser)/yaml/TrueHeading.yaml" />
  <param name="TrueHeading" value="3.1415" />

  <node ns="vehicle" pkg="sequoia_twist_controller" type="sequoia_twist_controller" name="twist_control" output="screen" />

  <node pkg="rviz" type="rviz" name="sequoia_nav_laser_rviz" respawn="true" args="-d $(find sequoia_nav_laser)/rviz/rviz_config.rviz" />

</launch>
